lightning/src/ln/channel.rs

```diff
@@ -2151,24 +2172,44 @@ impl<Signer: Sign> Channel<Signer> {
 			return Err(ChannelError::Ignore("Peer sent funding_locked when we needed a channel_reestablish. The peer is likely lnd, see https://github.com/lightningnetwork/lnd/issues/4006".to_owned()));
 		}
 
+		if let Some(scid_alias) = msg.short_channel_id_alias {
+			if Some(scid_alias) != self.short_channel_id {
```

Comment:
Should we sanitize further the scid alias ? Like ensuring it's not equal to the scid of other channels.

Also if we receive our counterparty's `funding_locked` before we have seen confirmation, the `short_channel_id` can be the temporary one. Then be replaced to the real one in `transactions_confirmed` ? It might be broken due to events unreliable ordering..

Reply:
> Like ensuring it's not equal to the scid of other channels.

That doesn't matter, we never see the SCID alias anywhere, we just shove it in the invoice and its up to our counterparty to figure it out.

> Also if we receive our counterparty's funding_locked before we have seen confirmation, the short_channel_id can be the temporary one

Not sure what you mean here - in that case `self.short_channel_id` will be `None` so the comparison will always be true.

Reply:
> Not sure what you mean here - in that case self.short_channel_id will be None so the comparison will always be true.

Eh right, though if we receive a `funding_locked` before we have seen the confirmation, we don't set `latest_inbound_scid_alias` and our peer might never resend a `funding_locked` letting this field as `None` ?

Reply:
No? If `self.short_channel_id` is `None` we'll always set the inbound scid alias field.